src/neo/Wallets/Wallet.cs

```diff
@@ -527,20 +527,25 @@ public long CalculateNetworkFee(DataCache snapshot, Transaction tx)
             int size = Transaction.HeaderSize + tx.Signers.GetVarSize() + tx.Attributes.GetVarSize() + tx.Script.GetVarSize() + IO.Helper.GetVarSize(hashes.Length);
             uint exec_fee_factor = NativeContract.Policy.GetExecFeeFactor(snapshot);
             long networkFee = 0;
+            int index = -1;
             foreach (UInt160 hash in hashes)
             {
+                index++;
                 byte[] witness_script = GetAccount(hash)?.Contract?.Script;
+                byte[] invocationScript = null;
 
-                if (witness_script is null && tx.Witnesses != null)
+                if (tx.Witnesses != null)
                 {
-                    // Try to find the script in the witnesses
-
-                    foreach (var witness in tx.Witnesses)
+                    if (witness_script is null)
                     {
-                        if (witness.ScriptHash == hash)
+                        // Try to find the script in the witnesses
+                        Witness witness = tx.Witnesses[index];
+                        witness_script = witness?.VerificationScript;
+
+                        if (witness_script is null)
```

Comment:
```suggestion
                        if (witness_script is null || witness_script.ToHexString().Equals(""))
```
Maybe should add this situation? I'm not sure how to make `witness_script` be null. I import a contract as watchonly address to my wallet and make it as sender, make it and another standard address as signers, then invoke a function from smart contract, make transaction relayed. Then I found the `witness_script` here is not null but byte[0].
![image](https://user-images.githubusercontent.com/38493437/113275583-80d37180-9311-11eb-8edb-009787aaf515.png)

Reply:
Fixed

Reply:
Nice catch!